Depot Fixes & Tweaks

Kyep opened the pull request:
Depot changes:
- **Fixed bug introduced by #11450 which caused depot mobs to aggro targets through walls in some situations, causing unavoidable and incorrect alert level rises. The depot is effectively unplayable without this fix.** Fixes #11737 
- Fixed another similar bug which was causing depot agents/officers AI's to completely switch off when they were not in active combat with a player, causing many systems (reporting in kills, radioing for backup against persistent foes, etc) to fail to work.
- Ported TG's [simple_human.dmi](https://github.com/tgstation/tgstation/blob/master/icons/mob/simple_human.dmi) which (A) cleanly separates 'human' simple mobs from animals (B) is consistent with TG (C) allows the use of a wider variety of syndi mob sprites (D) specifically allows depot officer mobs to have a 50% chance of being ranged laser mobs.... which makes cheesing the depot with plasmaglass windows a great deal harder.
- Converted depot to use plastitanium walls internally (like the lavaland base from TG does) and fixed the fact that plastitanium falsewalls did not smooth with those walls.
- Syndi agents can now get into the depot without fighting any mobs/turrets if they use the back door, and use an agent ID to disable the turrets.
- An oversight which caused syndie backup to instantly despawn upon seeing a fleeing space pod on red alert has been fixed.
- Bypassing a closed perimeter shield now triggers an alert level rise.
- The depot teleporter console now always creates bi-directional teleport portals. It also lets you choose the destination (out of all online teleport beacons). Lastly, it no longer auto-activates in peaceful mode. The idea of this is that an admin can create a teleport beacon somewhere, and then use the teleport depot computer to create a bi-directional gate into/out of the depot.
- The depot comms console no longer requires special access to do a one-time message to syndi HQ. Use this as a non-syndie at your peril.
- Lavaland visitors who attempt to sign in as depot visitors are now told that they're already recognized as syndies, instead of being told they're not recognized as syndies.
- The depot AI core, R.O.D.G.E.R, now has universal speak/understand so it can be used to talk to depot visitors when an admin takes control of it.
- Added ranged variants of officers and QM. These are designed to counter the "hide behind a glass window and shoot through it at the syndis" strategy. Tweaked balance a bit to account for this.
- Added a low (20%) chance of getting a stetchkin early on, both because its a weapon to help you against the mobs, and also because as a bullet weapon it doesn't support cheese strategies like a laser does.

Edited depot:
![depot_no_shuttle](https://user-images.githubusercontent.com/16434066/64920718-3c719680-d7aa-11e9-9496-b26baf21b770.png)



:cl: Kyep
fix: Fixed bug that made the syndicate depot impossibly difficult.
tweak: Various improvements to syndi depot, such as use of plastitanium walls, and nerfs to some cheese strategies.
add: Ported TG's simple_human.dmi, a sprite file of humanoid sprites (including syndicates).
add: Added a few extra syndi sprites from FoS.
/:cl:

TDSSS commented:
Do lavaland syndies need to sign in as visitors for the depot npcs to be peaceful? Will they get shot as traitors if anyone triggers the 'betrayal' alarm by opening a locker?

Kyep commented:
Assuming an admin allows lavabase syndies to travel to the depot:

- They have the syndicate faction, which means NPCs, turrets, and even mines ignore them.
- This means they don't need to sign in as visitors, and can loot things from the depot freely, without triggering countermeasures. After all, they are recognized syndie employees, and they have admin permission to be there.
- The area isn't 100% risk free for them, though. There are still one or two traps that are indiscriminate, and can hurt them. They'll have to avoid letting their guard down totally.

TDSSS commented:
Awesome. It will definitely make the syndie bases 'feel' more intelligent and dynamic. I like this idea a lot. Maybe at some point the lavaland base could get a communication console or emagged fax so they can contact their superiors easier? As far as I know the only current way for them to do so is the pretty immersion-breaking ahelp or prayers.

Kyep commented:
Issue with granting them a fax machine is that they can use it to fax any station department, which sorta goes against their goal of remaining hidden. A special fax machine that can ONLY fax the syndicate might be appropriate, though.

TDSSS commented:
I mean, they get comms, they could also just talk on comms, right? (unless they have some kind of special radios that can only talk in syndie comms?) If they contact the station they get banned or at least bwoinked.

TDSSS commented on code/modules/mob/living/silicon/decoy/decoy.dm:
```diff
@@ -30,6 +30,8 @@
 	icon_state = "ai-magma"
 
 /mob/living/silicon/decoy/syndicate/depot
+	universal_speak = 1
```
Shouldn't this be TRUE, same for understand?

Kyep replied:
Changed

daylight2 commented:
The syndicate communications officer literally has a objective to *sow seeds of chaos among the crew*, along with the capibilaty to copy any crewmembers voice and ID over radio, so it shouldn't be a concern of them 'revealing their presance'  via fax. AFAIK faxes don't even show which machine they were sent from, so it should be perfect! Perhaps even add a chameleon stamp to their equipment so they can send fake faxes to the station.

Kyep commented:
Guys, this PR is just about the shuttle and lavaland base / depot integration.

More general lavaland / lavabase stuff, such as changes to their comms, faxes, etc would be a different PR.

TDSSS commented:
You're right. With the new map format map merge conflicts aren't _that_ bad anymore, so I went and made my own PR for the fax stuff.

Kyep commented:
Updates made (all of these are listed in the PR description, with updated screenshot):
- Fixed bug causing depot mobs to aggro through walls.
- Fixed plastitanium falsewalls not smoothing with nodiagonal plastitanium walls.
- Ported TG simple_human.dmi sprite file
- Gave depot officers are 50% chance to be the new burst-laser variant, making it much harder to cheese the depot using reinforced windows
- Converted depot to using plastitanium walls like the TG lava base does, so they look consistent

Kyep commented:
Minor update:
Using ";" on the lavaland syndies' radio headsets now talks on syndicate frequency, not common. This prevents them from mistakenly outing themselves on radio, helps keep their base a secret (serving their objectives), and makes them less likely to interfere with main station. If doesn't prevent them talking to main station if they really want to, (illicit radio in telecomms room still can), but it does end the ridiculous situation where the secret syndie agents in the secret syndie base have their radios set to.... NT public radio by default.

Kyep commented:
Update: 

Tweaks to syndie lavabase:
- Fixed a bug that caused lavabase syndies' headsets to fail to spawn. 
- The syndicate comms agent can now only spawn in modes that contain traitors, such as traitor and autotraitor. This ensures they're never left without people to talk to / work with.
- The syndicate bioweapon scientists on lavabase can now only talk to people on main station using syndie radio. They can no longer talk on common radio, nor can they use the PDA message console. Those two things are now exclusive to the comms agent. To enforce this, the public access 'illicit intercom' has also been removed from telecomms (comms agent can still use their headset). These changes prevent the scientists from stealing the comms agent's job, accidentally giving themselves away using ";", and just generally trolling comms.

Couls commented on code/modules/ruins/lavalandruin_code/syndicate_base.dm:
```diff
@@ -63,19 +64,22 @@
 /obj/effect/mob_spawn/human/lavaland_syndicate/comms/space
 	flavour_text = "<span class='big bold'>You are a syndicate agent,</span><b> assigned to a small listening post station situated near your hated enemy's top secret research facility: Space Station 13. <b>Monitor enemy activity as best you can, and try to keep a low profile. Do not abandon the base.</b> Use the communication equipment to provide support to any field agents, and sow disinformation  ...
-	<br><i>You are free to attack anyone not aligned with the Syndicate in the vicinity of your base. <font size=6>DO NOT</font> work against Syndicate personnel (such as traitors or nuclear operatives). You may work with or against non-Syndicate antagonists on a case-by-case basis. <font size=6>DO NOT</font> leave your base without admin permission.</i>" 
+	<br><i>You are free to attack anyone not aligned with the Syndicate in the vicinity of your base. <font size=6>DO NOT</font> work against Syndicate personnel (such as traitors or nuclear operatives). You may work with or against non-Syndicate antagonists on a case-by-case basis. <font size=6>DO NOT</font> leave your base without admin permission.</i>"
```
Maybe add a line about keeping your presence hidden

Ralta replied:
It does say `try to keep a low profile`, so I guess that's reasonably clear.

Kyep replied:
Yeah. I may tweak it but for now "try to keep a low profile" is .... enough I think.

Couls commented on code/modules/ruins/lavalandruin_code/syndicate_base.dm:
```diff
@@ -63,19 +64,22 @@
 	. = ..()
-	if(prob(90)) //only has a 10% chance of existing, otherwise it'll just be a NPC syndie.
-		new /mob/living/simple_animal/hostile/syndicate/ranged(get_turf(src))
+	if(!GAMEMODE_IS_TRAITOR || prob(90)) // must be a mode with traitors, and roll 10% chance, to allow this
```
This is a 90% chance to roll if the gamemode is not traitors, you want prob(10) for a ten percent chance the previous line rolled 90% chance to spawn just an NPC

Kyep replied:
I *think* you're misreading it. It should "return INITIALIZE_HINT_QDEL" 100% of the time in non-traitor game modes, and 90% of the time in traitor modes. So it only persists (and thus, is active) 10% of the time in traitor modes.
The logic is equivalent to:
if (!GAMEMODE_IS_TRAITOR)
 return INITIALIZE_HINT_QDEL
if (prob(90))
 return INITIALIZE_HINT_QDEL
The INITIALIZE_HINT_QDEL means the object is deleted and not available for use by players.
If I'm wrong, let me know, but that's how it should (and I think, does) work.

Couls replied:
Right, I misread the probability part but i think you still want to use &&(and) instead of ||(or) because if it’s traitors it will be there 100%, but only have a 10% chance to spawn on non-traitor rounds(only one of the values have to be true for it to run the code underneath the or both values would need to be true if you used and which in this situation I think you want)

Kyep replied:
Right. Syndie comms agents shouldn't spawn at all if there are no traitors (since they have nobody to work with) and the code now enforces this.

TDSSS commented:
Might want to also allow comms agent to spawn during nukie rounds. They can watch cameras and give the nukies intel for example.

Kyep commented:
@TDSSS I strongly considered that, but ultimately decided against it due to how risky it would be. Specifically, if the comms agent can only spawn in traitor, autotraitor, or nuke ops.... and they talk on comms / out themselves, that means the station knows there is a 30%+ chance of nuke ops. Since nuke ops already almost always lose when they go loud... this means a single mistake by the comms agent can effectively ruin the round of the entire nuke ops team. I thought that would result in a lot of salt/anger by the nuke ops team. They'd be more likely to hurt the nuke ops team than to help them. So I left it disabled during nuke ops. Nuke ops can already pay TC for on-station traitors to help them out. I thought it best to leave 'backup' spawns under the nuke ops control, rather than potentially giving them a random chance to occur naturally, potentially outing the ops and ruining their mission.

TDSSS commented:
>Nuke ops can already pay TC for on-station traitors to help them out.

Wait what? Uh, wasn't this removed again since everyone hated it?

Kyep commented:
Not as far as I know, but my knowledge of that may be out of date. Regardless of whether it was or not, I am still skeptical that allowing a comms agent (who need only make one mistake on comms to alert the crew to syndies and make the nukies' mission more difficult) is a good idea during nukie rounds.

TDSSS commented:
Sleeper agents were added in #10357 which was testmerged but never merged. Onto the advantages of comms agents to nukies:

To be clear, I'm far from set on having the agent be there on nukie rounds. But

>If they're there, and good, they can.... use cameras? PDA? Minor benefits, considering nukies have pinpointers to locate the disk anyway. 

Since you're mostly concerned about stealth nukies having their cover blown, let's talk mostly about stealth nukies:
IMO, having full access to station cameras is a pretty big advantage, _especially_ when being stealthy. Sure, pinpointer can tell you 'captain is somewhere in the center of the station', or whatever, but someone watching cameras can tell you stuff like 'captain just entered his office, he's alone' or 'watch out, blueshield coming his way'. Remember that the comms agent camera console works like AI view, where you can freely pan around and see everything the AI sees.

Accidentally talking on common and giving things away is a threat with nukies themselves as well. You already have 5 people capable of accidentally giving everything away with one wrong ';', and in the case of the comms agent it's not even guaranteed.

When nukies do go loud, camera overwatch is still very useful. Knowing where stuff like mechs are, or hw many people are protecting the captain, or what kind of barricades crew has built.

I admit, I haven't personally seen any comms agents screw up and give their presence away by accident, so maybe that's more of an issue than I think it is.

Kyep commented:
The general feedback I've gotten is that we should just outright remove comms agents.
I'm not sure I want to go that far yet, but I do think they should be heavily curtailed so that they'll only spawn in rounds where they might have a use.
I'm skeptical that they're worth having in nukie rounds, for the above reasons. So, for now, they don't spawn in those.

Markolie commented:
The Syndicate base tweaks in this pull request should really be moved to a separate pull request. These are much more controversial changes and added after the original creation of the PR (AFAIK) and as such not everybody is aware of them. This is especially true with the new Lavaland ruins being in, which very significantly reduces the odds of the base spawning.

Kyep commented:
@Markolie I'm skeptical about both of those points. I'm skeptical that they're more controversial, based on the reactions I have got from people. I'm also skeptical that people aren't aware of them - they were marked in a reply and in the PR description. If I made a whole PR just for the comms agent tweaks, I might be inclined to just remove comms agent completely - it is essentially "trolling - the job" and serves little purpose except to annoy the crew and give away the syndicate's presence.

Markolie commented:
Feedback you received does not reflect the opinion of the whole server. Additionally, with more than a hundred PR’s open, it’s nearly impossible to keep track of everything in the repository. Only those who commented/subscribed would be aware of a reply. As the person who added the base, I strongly urge you to make a separate pull request. Feel free to turn it into a removal PR if you so desire.

Kyep commented:
I'll think about it. 
I was also thinking about splitting this PR in two. One for the depot stuff, one for everything else.
I was also thinking about just scrapping the syndi shuttle, or maybe even all lavabase changes, and only keeping the depot stuff.
I'll think about it some more before I decide what to do.

Kyep commented:
Updated to fix merge conflict.
I will split out the comms agent / headset stuff into another PR if any head or maint asks me to. Thus far, none has asked me to.

Kyep commented:
**Major update:**
- Syndicate shuttle removed from this PR. Description updated accordingly. The remaining PR content focuses on fixes/tweaks to syndi depot and (to a much lesser extent) lavabase.

Kyep commented:
@JKnutson101 It doesn't change the spawn rate of the syndicate lavaland base, or of the regular syndi scientist roles there. The only thing it changes the spawn rate of is the syndi comms agent on lavabase, and even then it does so to ensure that they only spawn in rounds with traitors, thus, rounds where they can have someone else to talk to / work with.

Fox-McCloud commented:
Alright, the lavaland syndi base changes should definitely be split off into their own PR, as it's a completely separate things from this (and isn't part of a larger refactor, but an intentional specific balance/design change).

Fox-McCloud commented on code/modules/mob/living/simple_animal/hostile/syndicate.dm:
```diff
@@ -200,10 +207,23 @@
 	alert_on_death = TRUE
 	melee_block_chance = 60
 
+/mob/living/simple_animal/hostile/syndicate/melee/autogib/depot/officer/Initialize()
```
needs mapload

Kyep replied:
Done

Fox-McCloud commented on code/modules/mob/living/simple_animal/hostile/syndicate.dm:
```diff
@@ -200,10 +207,23 @@
 	melee_block_chance = 60
 
+/mob/living/simple_animal/hostile/syndicate/melee/autogib/depot/officer/Initialize()
+	..()
```
need to do a non-explicit return here that still returns parent value.

Kyep replied:
Done

Kyep commented:
Updated to fix merge conflict.

Kyep commented:
Updated to fix merge conflict.

Kyep commented:
Fixed merge conflict.

Fox-McCloud commented:
We're in desperate need of sprite splits, so thank you.

variableundefined commented:
Ping when ready for review please.

Fox-McCloud commented on code/game/objects/structures/false_walls.dm:
```diff
@@ -326,7 +326,7 @@
 	walltype = /turf/simulated/wall/mineral/plastitanium
 	smooth = SMOOTH_MORE
-	canSmoothWith = list(/turf/simulated/wall/mineral/plastitanium, /obj/machinery/door/airlock/shuttle, /obj/machinery/door/airlock, /obj/structure/window/full/shuttle, /obj/structure/shuttle/engine/heater)
+	canSmoothWith = list(/turf/simulated/wall/mineral/plastitanium, /turf/simulated/wall/mineral/plastitanium/nodiagonal, /obj/machinery/door/airlock/shuttle, /obj/machinery/door/airlock, /obj/structure/window/full/shuttle, /obj/structure/shuttle/engine/heater)
```
Uhhhh, I'm pretty sure that this works with subtypes?

Kyep replied:
I just tested it, and opening/closing the falsewall without this change causes it to fail to smooth properly with the surrounding walls.

Kyep commented:
Updated:
- Fixed false walls next to bathroom not smoothing properly. (Separate issue from Fox's comment)
- Added ranged variants of officer and QM mobs, specifically to counter the "hide behind a glass window and shoot them with a laser through the glass" strategy I've seen employed in the depot. The ranged variants all shoot through glass, and will attack you through your cover.
- Added a new type of loot random spawner on tables, which gives raiders a 20% chance to obtain a stetchkin early on. You still have to fight your way past at least one turret to get this, and you only have a 20% chance per table spawner you reach to get it, but at least it offers a hope of getting a weapon to help you early on, and hopefully reduces the reliance on cheese strats a bit.

Kyep commented:
@variableundefined Ready for review.

variableundefined commented:
Squash Merging
